File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length-1;
        for (int i = 0; i < numbers.length; i++) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                return new int[]{left+1, right+1};
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
        throw new IllegalArgumentException();
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var left = 0
        var right = numbers.size-1
        for (i in 0 until numbers.size) {
            val sum = numbers[left] + numbers[right]
            if (sum == target) {
                return intArrayOf(left+1, right+1)
            } else if (sum < target) {
                left++
            } else {
                right--
            }
        }
        throw Exception()
    }
}